pkg/components/components.go

```diff
@@ -368,63 +368,198 @@ func GetClusterRole() *rbacv1.ClusterRole {
 	return role
 }
 
-func GetCrd() *extv1beta1.CustomResourceDefinition {
-	crd := &extv1beta1.CustomResourceDefinition{
+func GetCrd() *extv1.CustomResourceDefinition {
```

Comment:
Don't we want to annotate the _types with descriptions and generate this with gen-openapi ?

Reply:
the existence of the `/shared` folder in types cause an error with `operator-sdk generate crds` command.
so, I created  it in an older version for reference and then recreated it in components.go.
the resulting .crd file is identical (except for the version migration changes of course)